.prettierrc

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
since prettier is not installed in `package.json`, is this config necessary?

Reply:
it's just for whoever uses vscode with prettier to have constant formatting. I suggest to add a `CONTRIBUTE.md` and make it compulsory to install the prettier. 
but you also have a point. I move whatever I can to .editorconfig. but some stuff isn't supported by editor config that I will keep here

Reply:
ok so could you install prettier locally?

and yes, I suspect it is time to setup `CONTRIBUTE.md` (and some proper tests) 😅

Reply:
actually it doesn't really matter, it's just an instruction to the vscode's prettier. just wanted to get the same output as your current formatting. if I add it to the package.json you will force all the files get formatted which aren't related to this PR.

Reply:
this is I wanted to avoid 

![image](https://user-images.githubusercontent.com/8991783/106699585-1e663980-661e-11eb-8fc2-414d5b47bd5e.png)

I just ran formatting and lint over the lib and whole lib is affected. should I push it?

Reply:
yes sure! I mean if we are adding linting to the project, we should add it properly, which does involve a change to all existing code. of course it would be better if there were separate PRs for these two changes (the bugfix and the linting), but since the scope of the bug-change is limited it is perfectly fine.

Reply:
sure I will push it, but if you want to add linting and formatting, I suggest using `nx`. it's battery included